tracer/src/Datadog.Trace.ClrProfiler.Native/method_rewriter.cpp

```diff
@@ -218,61 +225,95 @@ HRESULT TracerMethodRewriter::Rewrite(RejitHandlerModule* moduleHandler, RejitHa
+                    if (argTypeFlags & TypeFlagByRef)
+                    {
+                        Logger::Warn("*** CallTarget_RewriterCallback(): Methods with ref parameters "
+                                     "cannot be instrumented. ");
+                        return S_FALSE;
+                    }
                 }
             }
-            else
+        }
+        else
+        {
+            // Load the arguments inside an object array (SlowPath)
+            reWriterWrapper.CreateArray(tracerTokens->GetObjectTypeRef(), numArgs);
+            for (int i = 0; i < numArgs; i++)
             {
+                reWriterWrapper.BeginLoadValueIntoArray(i);
                 reWriterWrapper.LoadArgument(i + (isStatic ? 0 : 1));
+                const auto [elementType, argTypeFlags] = methodArguments[i].GetElementTypeAndFlags();
                 if (argTypeFlags & TypeFlagByRef)
                 {
                     Logger::Warn("*** CallTarget_RewriterCallback(): Methods with ref parameters "
                                  "cannot be instrumented. ");
                     return S_FALSE;
                 }
+                if (argTypeFlags & TypeFlagBoxedType)
+                {
+                    const auto& tok = methodArguments[i].GetTypeTok(metaEmit, tracerTokens->GetCorLibAssemblyRef());
+                    if (tok == mdTokenNil)
+                    {
+                        return S_FALSE;
+                    }
+                    reWriterWrapper.Box(tok);
+                }
+                reWriterWrapper.EndLoadValueIntoArray();
             }
         }
     }
     else
     {
-        // Load the arguments inside an object array (SlowPath)
-        reWriterWrapper.CreateArray(tracerTokens->GetObjectTypeRef(), numArgs);
-        for (int i = 0; i < numArgs; i++)
-        {
-            reWriterWrapper.BeginLoadValueIntoArray(i);
-            reWriterWrapper.LoadArgument(i + (isStatic ? 0 : 1));
-            const auto [elementType, argTypeFlags] = methodArguments[i].GetElementTypeAndFlags();
-            if (argTypeFlags & TypeFlagByRef)
-            {
-                Logger::Warn("*** CallTarget_RewriterCallback(): Methods with ref parameters "
-                             "cannot be instrumented. ");
-                return S_FALSE;
-            }
-            if (argTypeFlags & TypeFlagBoxedType)
-            {
-                const auto& tok = methodArguments[i].GetTypeTok(metaEmit, tracerTokens->GetCorLibAssemblyRef());
-                if (tok == mdTokenNil)
-                {
-                    return S_FALSE;
-                }
-                reWriterWrapper.Box(tok);
-            }
-            reWriterWrapper.EndLoadValueIntoArray();
-        }
+        // Load the methodDef token to produce a RuntimeMethodHandle on the stack
+        reWriterWrapper.LoadToken(caller->id);
+
+        runtimeMethodHandleBuffer[0] = ELEMENT_TYPE_VALUETYPE;
```

Comment:
this block.

Reply:
I can move the COR_SIGNATURE's into here, but I wasn't sure if there would be scoping issues since the buffer address is used below in the `pbBase`. I'm happy to try any suggestions you have

Reply:
If it's okay with you, I would like to keep this as-is and this could be refactored later. Is that okay?

Reply:
totally ok, resolving this.